sim_pipeline/gg_lens_pop.py

```diff
@@ -66,22 +66,19 @@ def draw_population(self):
         :return: List of GGLens instances with parameters of the deflectors and lens and source light.
         :rtype: list
         """
-        valid_lenses = []
-        count = 0
 
-        num_trials = 10000  #need implement
-        for _ in range(num_trials):
-            # using draw_galaxy and draw_deflector randon lensing system
+        # Initialize an empty list to store the GGLens instances
+        gg_lens_population = []
+        # Estimate the number of lensing systems
+        num_lenses = self._lens_galaxies.deflector_number()
+        num_sources = self._source_galaxies.galaxies_number()
+        num_lens_systems = int(num_lenses * num_sources)
```

Comment:
these are the permutations of possibilities, not the reality of the number of actual lens systems